desc/continuation.py

```diff
@@ -167,46 +104,283 @@ def solve_continuation_automatic(  # noqa: C901
+            deltas = {}
+
+        stop = not eqi.is_nested()
+
+        if not stop:
+            eqi.solve(
+                optimizer=optimizer,
+                objective=objective_i,
+                constraints=constraints_i,
+                ftol=ftol,
+                xtol=xtol,
+                gtol=gtol,
+                verbose=verbose,
+                maxiter=nfev,
+            )
+        stop = stop or not eqi.is_nested()
+        eqfam.append(eqi)
+
+        if checkpoint_path is not None:
+            if verbose > 0:
+                print("Saving latest iteration")
+            eqfam.save(checkpoint_path)
+        timer.stop("Iteration {} total".format(ii + 1))
+        if verbose > 1:
+            timer.disp("Iteration {} total".format(ii + 1))
+        ii += 1
+
+    if stop:
+        if mres_step == MIN_MRES_STEP:
+            raise RuntimeError(
+                "Automatic continuation failed with mres_step=1, "
+                + "something is probaby very wrong with your desired equilibrium."
+            )
+        else:
+            warnings.warn(
+                colored(
+                    "WARNING: Automatic continuation failed with "
+                    + f"mres_step={mres_step}, retrying with mres_step={mres_step/2}",
+                    "yellow",
+                )
+            )
+            return _solve_axisym(
+                eq,
+                mres_step // 2,
+                objective,
+                optimizer,
+                pert_order,
+                ftol,
+                xtol,
+                gtol,
+                nfev,
+                verbose,
+                checkpoint_path,
+            )
+
+    return eqfam
+
+
+def _add_pressure(
+    eq,
+    eqfam,
+    pres_step,
+    objective="force",
+    optimizer="lsq-exact",
+    pert_order=2,
+    ftol=None,
+    xtol=None,
+    gtol=None,
+    nfev=100,
+    verbose=1,
+    checkpoint_path=None,
+):
+    """Add pressure with adaptive step sizing."""
+    timer = Timer()
+
+    eqi = eqfam[-1].copy()
+    eqfam_temp = eqfam.copy()
+    # make sure its at full radial/poloidal resolution
+    eqi.change_resolution(L=eq.L, M=eq.M, L_grid=eq.L_grid, M_grid=eq.M_grid)
+
+    constraints_i = get_fixed_boundary_constraints(
+        iota=objective != "vacuum" and eq.iota is not None,
+        kinetic=eq.electron_temperature is not None,
+    )
+    objective_i = get_equilibrium_objective(objective)
+
+    pres_steps = (
+        0
+        if (abs(eq.pressure(np.linspace(0, 1, 20))) < 1e-14).all() or pres_step == 0
+        else int(np.ceil(1 / pres_step))
+    )
+    pres_ratio = 0 if pres_steps else 1
+
+    ii = len(eqfam_temp)
+    stop = False
+    while ii - len(eqfam_temp) < pres_steps and not stop:
+        timer.start("Iteration {} total".format(ii + 1))
+        # increase pressure
+        deltas = get_deltas(
+            {"pressure": eqfam_temp[-1].pressure}, {"pressure": eq.pressure}
+        )
+        deltas["p_l"] *= pres_step
```

Comment:
does this ensure the steps are equal in pressure?
if desired pressure is `p1` and original pressure is `0` and pres_steps=4

step 1: `deltas["p_l"]` = 0.25 * (p1 - 0)...p = 0.25 p1
step 2: `deltas["p_l"]` = 0.25 * (p1 - 0.25*p1) = 3/16 * p1 ... p = 7/16 p1
step 3: `deltas["p_l"]` = 0.25 * (p1 - 3/16*p1) = 13/64 * p1 ... p =41/64 p1
step 4: `deltas["p_l"]` = 0.25 * (p1 - 41/64*p1) = 41/256 * p1 ... p != p1 

I must be misunderstanding this...

Reply:
ah nevermind, eqfam_temp is not added to during this loop, so the deltas are all just 0.25 * p1, I see now